### PR TITLE
Add OpenSea apiKey to headers

### PR DESCRIPTION
**Pull request by avonian**
@raymondjacobson submitting a PR that adds the OpeanSea apiKey to the request headers

### CODE REVIEW

File: src/eth/index.ts [hunk 1]
```diff
@@ -62,20 +62,24 @@ export class OpenSeaClient {
...
+  private fetchOptions = {};
 
   constructor(props?: OpenSeaClientProps) {
     this.url = props?.apiEndpoint ?? this.url
     this.apiKey = props?.apiKey ?? this.apiKey
     this.assetLimit = props?.assetLimit ?? this.assetLimit
     this.eventLimit = props?.eventLimit ?? this.eventLimit
+    if(this.apiKey) {
+      this.fetchOptions['X-API-KEY'] = this.apiKey
+    }
```
Comment (Kyle-Shanks) [inline src/eth/index.ts:74]:
```suggestion
  if(this.apiKey) {
    this.fetchOptions = {
      headers: new Headers({ 'X-API-KEY': this.apiKey })
    }
  }
```

Small change: api key needs to be set to a header within a headers prop on the request options

Comment (Kyle-Shanks) [reply src/eth/index.ts:74]: Will make a new PR to get this out today. Closing in favor of https://github.com/AudiusProject/fetch-nft/pull/16

**Comment by Kyle-Shanks** [comment]
Closing in favor of https://github.com/AudiusProject/fetch-nft/pull/16

File: src/eth/index.ts (hunk 1 above)
Comment (avonian) [reply src/eth/index.ts:74]: oh yea good catch lol, i did a last second refactor and evidently tripped over myself :sweat_smile:

**Pull request closed without merging**